### PR TITLE
Moved gossipsub block broadcast to after all of the artifact is created.

### LINKED ISSUE #1246: Artifact p2p provide should happen before new block broadcast
Current situation:

A build node starts a build for a new artifact. When the build succeeds, it follows this procedure:

1. it creates the transparency log for the artifact and uses the transparency log service to add it to the ledger
2. the transparency log service uses the blockchain service to add the log as a payload to a new block
 (this is where the consensus should happen)
3. when the block is added, it is broadcast to connected peers, who store the transparency log in their local log database
4. after this, the log is written to the build node's transparency log database
5. and only after this, the artifact file is provided on the p2p network

The problem with this:

- step 3 is awaits the broadcast of new blocks, so step 4 and 5 only happens when the broadcast is done
- as a result, a peer that receives the block early on, can have a transparency log for a given artifact, but it won't be able to download it because step 5 didn't execute yet

Possible solution:

- the broadcast of new blocks should not block the publication process
- explore better ways for this mechanism to work, and we should include the consensus mechanism in our thinking because this might change the way this works anyway

### PR DESCRIPTION
**Pull request by efrisch**
## Description

Fixes pyrsia/pyrsia#1246

This PR delays broadcasting new blocks until after the artifact(s) are all added by making the broadcast a separate function, it takes a vec of payloads, and that happens after the loop adding all of the artifact component files.

## Screenshots (optional)
Integration Tests Run

## PR Checklist

- [x] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [x] I've read ["What is a Good PR?"](https://github.com/pyrsia/pyrsia/blob/main/docs/community/get_involved/good_pr.md)
- [x] I've included a good title and brief description along with how to review them.
- [x] I've linked any associated an [issue](https://github.com/pyrsia/pyrsia/issues).

### Code Contributions

- [x] I've run [pre-commit.sh](https://github.com/pyrsia/pyrsia/blob/main/pre-commit.sh) ([pre-commit.bat](https://github.com/pyrsia/pyrsia/blob/main/pre-commit.bat)) successfully.
- [x] I've run the [integrations tests](https://github.com/pyrsia/pyrsia-integration-tests#how-to-set-up-and-run-the-tests)
on the PR branch and all tests passes.

### CODE REVIEW

**Review by dasmanas — DISMISSED** [review]
lgtm 👌

File: src/transparency_log/log.rs [hunk 1]
```diff
@@ -231,17 +231,30 @@ impl TransparencyLogService {
     pub async fn add_artifact(
         &self,
         add_artifact_request: AddArtifactRequest,
-    ) -> Result<TransparencyLog, TransparencyLogError> {
+    ) -> Result<(TransparencyLog, String), TransparencyLogError> {
         let transparency_log = TransparencyLog::from(add_artifact_request);
 
         let payload = serde_json::to_string(&transparency_log)?;
+        /*
```
Comment (karolh2000) [inline src/transparency_log/log.rs:238]: probably it'd be better to remove this dead code? any particular reason why it wasn't removed?

**Review by karolh2000 — DISMISSED** [review]

File: src/transparency_log/log.rs (hunk 1 above)
Comment (efrisch) [reply src/transparency_log/log.rs]: Removed now.

**Review by karolh2000 — APPROVED** [review]

File: src/artifact_service/service.rs [hunk 2]
```diff
@@ -149,11 +150,13 @@ impl ArtifactService {
...
             );
 
-            let add_artifact_transparency_log = self
+            let add_artifact_transparency_tuple = self
                 .transparency_log_service
                 .add_artifact(add_artifact_request)
                 .await?;
 
+            let add_artifact_transparency_log = add_artifact_transparency_tuple.0;
+            payloads.push(add_artifact_transparency_tuple.1);
```
Comment (fishseabowl) [inline src/artifact_service/service.rs:159]: Looks good. But `add_artifact_transparency_tuple.0` and `add_artifact_transparency_tuple.1` seem confusing to me. Maybe structure is better.

**Review by dasmanas — APPROVED** [review]
lgtm

**Review by fishseabowl — COMMENTED** [review]
From your code, the build request workflow looks like this: when a build request is triggered, some artifacts are ready, then these artifacts are added to the transparency_log(`transparency_log_service.add_artifact`), next the swarm is notified that the node is a provider (`p2p_client.provide`), and finally these artifacts are added to the blockchain and broadcast (`transparency_log_service.broadcast_artifacts`). 
I'm not sure if I understand your code correctly. Could you please provide details on this workflow? Thanks

**Pull request merged**